__tests__/app.test.js

```diff
@@ -163,4 +163,104 @@ describe('GET /api/reviews/:review_id/comments', () => {
       expect(response.body.msg).toBe('bad request')
       })
   });
+});
+
+describe.only("/api/reviews/:review_id/comments", () => { 
+  test("Get status 201 response and an object containing the newly posted comment", () => {
+      const newComment = {
+        username: "mallionaire",
+        body: "giant Jenga is better",
+      };
+      return request(app)
+        .post("/api/reviews/2/comments")
+        .send(newComment)
+        .expect(201)
+        .then((response) => {
+          expect(response.body.comment.comment_id).toBe(7);
+          expect(response.body.comment.body).toBe("giant Jenga is better");
+          expect(response.body.comment.review_id).toBe(2);
+          expect(response.body.comment.author).toBe("mallionaire");
+          expect(response.body.comment.votes).toBe(0);
+          expect(typeof response.body.comment.created_at).toBe("string");
+          });
+  });
+  test("Return error 404 and message of 'no review found at this id!'", () => {
+   const newComment = {
+        username: "mallionaire",
+        body: "giant Jenga is better",
+      };
+  return request(app)
+  .post("/api/reviews/20/comments")
+        .send(newComment)
+        .expect(404)
+        .then((response) => {
+         expect(response.body.msg).toBe('no review found at this id!')
+        })
+  });
+  test("Return error 400 and message of 'bad request' if id is not a number", () => {
+   const newComment = {
+        username: "mallionaire",
+        body: "giant Jenga is better",
+      };
+  return request(app)
+  .post("/api/reviews/NotAnId/comments")
+        .send(newComment)
+        .expect(400)
+        .then((response) => {
+         expect(response.body.msg).toBe('bad request')
+        })
+  });
+  test("Return error 400 and message of 'bad request' if message body is missing properties", () => {
+    const newComment = {
+         username: "mallionaire",
+       };
+   return request(app)
+   .post("/api/reviews/2/comments")
+         .send(newComment)
+         .expect(400)
+         .then((response) => {
+          expect(response.body.msg).toBe('bad request')
+         })
+   });
+  test("Return error 400 and message of 'bad request' if message body has extra properties", () => {
+    const newComment = {
+         username: "mallionaire",
+         body: "giant Jenga is better",
+         votes: 7
```

Comment:
When extra properties are posted, this should be ignored, as per the ticket: **Note: the status code should be 201 which signifies that something was created**